src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -7252,6 +7243,58 @@ internal static PSSnapInInfo GetEngineSnapIn(ExecutionContext context, string na
 
             return null;
         }
+
+        /// <summary>
+        /// Get the last (UTC) write time for the file at the given path if it exists and return true.
+        /// If the file does not exist, return false.
+        /// </summary>
+        /// <param name="path">the path to the file with the last write time to check</param>
+        /// <param name="lastWriteTimeUtc">the last time the file was written to, as a UTC DateTime</param>
+        /// <returns>true if the file exists and the returned last write time is valid, false otherwise</returns>
+        internal static bool TryGetFileLastWriteTimeUtc(string path, out DateTime lastWriteTimeUtc)
+        {
+            try
+            {
+                lastWriteTimeUtc = File.GetLastAccessTimeUtc(path);
+                return true;
+            }
+            catch (Exception e)
+            {
+                ModuleIntrinsics.Tracer.WriteLine("Exception checking LastWriteTimeUtc on module {0}: {1}", path, e.Message);
+                lastWriteTimeUtc = DateTime.MinValue;
+                return false;
```

Comment:
We return `lastWriteTimeUtc` in any case. So I suggest rename to `GetFileLastWriteTimeUtc`, don't return a bool and directly return `GetFileLastWriteTimeUtc`
```c#
internal static DateTime GetFileLastWriteTimeUtc(string path) 
```